src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -357,8 +357,14 @@ private function dehydrateValue(mixed $value, LivePropMetadata $propMetadata, ob
             if (!interface_exists(NormalizerInterface::class)) {
                 throw new \LogicException(sprintf('The LiveProp "%s" on component "%s" has "useSerializerForHydration: true", but the Serializer component is not installed. Try running "composer require symfony/serializer".', $propMetadata->getName(), $parentObject::class));
             }
+            if (null === $this->serializer) {
+                throw new \LogicException(sprintf('The LiveProp "%s" on component "%s" has "useSerializerForHydration: true", but no serializer has been set.', $propMetadata->getName(), $parentObject::class));
+            }
+            if (!$this->serializer instanceof NormalizerInterface) {
+                throw new \LogicException(sprintf('The LiveProp "%s" on component "%s" has "useSerializerForHydration: true", but the given serializer does not implement NormalizerInterface.', $propMetadata->getName(), $parentObject::class));
+            }
```

Comment:
Do we need those two errors? How those two errors could happen?

Reply:
The null test is required (it's not because symfony/serializer is installed that it is configured in Symfony).

The instance test is necessary because the constructor argument is NormalizerInterface|DenormalizerInterface .. so one could be normalizer without beeing denormalizer :|